tests/flexalloc_ut_slab.c

```diff
@@ -50,101 +50,103 @@ test_slabs(const struct test_vals * test_vals)
+              test_vals->disk_min_lbs, fla_geo_slabs_lb_off(&fs->geo));
+  available_lb_for_slabs = test_vals->disk_min_lbs - fla_geo_slabs_lb_off(&fs->geo);
+  expected_slabs = available_lb_for_slabs / test_vals->slab_nlb;
 
   /* Test values in struct fla_slabs */
-  err |= FLA_ASSERTF(*fs->slabs.fslab_num == test_vals->nslabs,
-                     "Unexpected number of free slabs (%d == %d)", *fs->slabs.fslab_num, test_vals->nslabs);
-  err |= FLA_ASSERTF(*fs->slabs.fslab_head == 0,
-                     "Unexpected head ID (%d == %d)", *fs->slabs.fslab_head, 0);
-  err |= FLA_ASSERTF(*fs->slabs.fslab_tail == test_vals->nslabs - 1,
-                     "Unexpected tail ID (%d == %d)", *fs->slabs.fslab_tail, test_vals->nslabs - 1);
+  err |= FLA_ASSERTF(*fs->slabs.fslab_num == expected_slabs,
+                     "Unexpected number of free slabs (%d == %d)",
+                     *fs->slabs.fslab_num, expected_slabs);
+
+  /*err |= FLA_ASSERTF(*fs->slabs.fslab_head == 0,
+                     "Unexpected head ID (%d == %d)", *fs->slabs.fslab_head, 0);*/
+
+  err |= FLA_ASSERTF(*fs->slabs.fslab_tail == expected_slabs - 1,
+                     "Unexpected tail ID (%d == %d)",
+                     *fs->slabs.fslab_tail, expected_slabs - 1);
 
   /* Acquire all the slabs and then release them all */
-  for(uint32_t slab_offset = 0 ; slab_offset < test_vals->nslabs ; ++slab_offset)
+  for(uint32_t slab_offset = 0 ; slab_offset < expected_slabs ; ++slab_offset)
   {
     slab_header = (void*)fs->slabs.headers + (slab_offset * sizeof(struct fla_slab_header));
 
     err = fla_acquire_slab(fs, test_vals->obj_nlb, &slab_header);
     if(FLA_ERR(err, "fla_acquire_slab()"))
-    {
       goto close_fs;
-    }
 
     err = FLA_ASSERT(slab_header->next == FLA_LINKED_LIST_NULL,
                      "Next pointer is not null after slab format");
     err |= FLA_ASSERT(slab_header->prev == FLA_LINKED_LIST_NULL,
                       "Prev pointer is not null after slab format");
     if(FLA_ERR(err, "FLA_ASSERT()"))
-    {
       goto close_fs;
-    }
 
-    const uint32_t free_slabs = test_vals->nslabs - (slab_offset + 1);
+    const uint32_t free_slabs = expected_slabs - (slab_offset + 1);
     err = FLA_ASSERTF(*fs->slabs.fslab_num == free_slabs,
-                      "Unexpected number of free slabs (%d == %d)", *fs->slabs.fslab_num, free_slabs);
+                      "Unexpected number of free slabs (%d == %d)",
+                      *fs->slabs.fslab_num, free_slabs);
     if(FLA_ERR(err, "FLA_ASSERTF()"))
-    {
       goto close_fs;
-    }
 
     err = test_check_slab_pointers(fs, free_slabs);
     if(FLA_ERR(err, "test_check_slab_pointers()"))
-    {
       goto close_fs;
-    }
   }
 
   /* If we acquire another slab, we should receive an error */
   ret = fla_acquire_slab(fs, test_vals->obj_nlb, &slab_error);
   err = FLA_ASSERT(ret != 0, "Acquire of an empty free list did NOT fail");
   if(FLA_ERR(err, "FLA_ASSERT()"))
-  {
```

Comment:
Changing formatting is more tangential, but it does raise a question. We discussed whether omitting braces would be our preferred approach before.

Can we revisit and perhaps always have braces ?

https://discord.com/channels/778163622324338688/819088333682507796/938422807803625533

https://discord.com/channels/778163622324338688/819088333682507796/938480300156137565

https://discord.com/channels/778163622324338688/819088333682507796/938423009302175776

?

Reply:
This always bites me in the but. Let me fix that real quick :). thx for bringing this up.